src/ui/App.tsx

```diff
@@ -19,9 +19,34 @@ export interface ThreadActions {
   keep: (messages: Message[]) => Promise<Response>;
 }
 
+interface ThreadsState {
+  threads: gapi.client.gmail.Thread[];
+}
+
+export interface UpdateThreadListAction {
+  removeThreadId?: string;
+  threads?: gapi.client.gmail.Thread[];
+}
+
 function App(): JSX.Element {
-  const [threadIndex, setThreadIndex] = useState(0);
-  const [threads, setThreads] = useState([] as gapi.client.gmail.Thread[]);
+  function reducer(
```

Comment:
Is it worth verifying that only one of removeThreadId or threads is populated?

Reply:
Or is there a case where it's reasonable to populate both?

Reply:
Currently, there's no case the populates both. A problem with post-facto code reviews is addressing small things like this is possibly more work than it's worth. Not sure.